### PR TITLE
Moved and renamed types files in a common types folder

### PR DESCRIPTION
**Pull request by thehenrytsai**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # Decentralized Web Node (DWN) SDK
 
 Code Coverage
-![Statements](https://img.shields.io/badge/statements-94.65%25-brightgreen.svg?style=flat) ![Branches](https://img.shields.io/badge/branches-94.16%25-brightgreen.svg?style=flat) ![Functions](https://img.shields.io/badge/functions-93.06%25-brightgreen.svg?style=flat) ![Lines](https://img.shields.io/badge/lines-94.65%25-brightgreen.svg?style=flat)
```
Comment (diehuxx) [inline README.md:6]: nice

Comment (diehuxx) [inline README.md:6]: nice

**Review by diehuxx — COMMENTED** [review]
Was the intent just to move all `types.ts` files into a common `types/` folder or also to migrate the types defined in files other than the many `types.ts`? E.g. `dwn.ts` defines the `DwnConfig` type.

**Comment by thehenrytsai** [comment]
In my mind, min-bar is that if you were gonna go out of your way to create a file that contains types, put it in somewhere common and name it accordingly.

A separate discussion can be had about whether we want to enforce types to ALWAYS to be defined in a separate file in the types folder, that seems more contentious and can be in a follow up PR (can also be in this PR if you prefer also).

**Review by diehuxx — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (thehenrytsai) [reply README.md:6]: 🙏

**Pull request merged**